src/pools/D3M4626TypePool.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-FileCopyrightText: © 2021 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPool.sol";
+import "forge-std/interfaces/IERC4626.sol";
+
+interface VatLike {
+    function live() external view returns (uint256);
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface IUniversalRewardsDistributor {
+    function claim(address account, address reward, uint256 claimable, bytes32[] memory proof)
+        external
+        returns (uint256 amount);
+}
+
+contract D3M4626TypePool is ID3MPool {
+    /* EVENTS */
+
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, address data);
+
+    /* CONSTANTS */
+
+    IERC20 public immutable dai;
+    IERC4626 public immutable vault;
+    VatLike public immutable vat;
+
+    /* STORAGE */
+
+    address public hub;
+    mapping(address => bool) public wards;
```

Comment:
`wards` should be `uint256` instead of `bool` to match other interfaces

Reply:
Both are interchangeable, but if you prefer like that I change